src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -887,7 +887,7 @@ public static boolean addItemStackToProvider(@NotNull final ICapabilityProvider
      *
      * @param provider  {@link ICapabilityProvider} to add itemstack to.
      * @param itemStack ItemStack to add.
-     * @return True if successful, otherwise false.
+     * @return Empty when successful, otherwise return the itemStack which was supposed to be added.
```

Comment:
This is not correct.
It Returns the stack it replaced. Which might be the original stack if nothing is replaced, however this is not guaranteed.